### PR TITLE
[CI][Inductor] Skip CPU tests when running on GPU

### LINKED ISSUE #115423: `PYTORCH_TESTING_DEVICE_ONLY_FOR=cuda` not affecting inductor unit tests
### 🐛 Describe the bug

`PYTORCH_TESTING_DEVICE_ONLY_FOR=cuda` is usually a great way to reduce the number of unit tests executed in a suite when trying to debug issues specific to GPUs. It has been observed that this environment variable is not having the desired affect for inductor based unit tests, this results in excess CI workload in which the cpu inductor unit tests are executed in many different workflows.

Example:
```
PYTORCH_TESTING_DEVICE_ONLY_FOR=cuda pytest test/inductor/test_torchinductor.py --verbose
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_broadcast1 PASSED [0.4896s]                                                            [  0%]
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_broadcast2 PASSED [0.0574s]                                                            [  0%]
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_broadcast3 PASSED [0.0518s]                                                            [  0%]
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_dense PASSED [0.0624s]                                                                 [  0%]
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_double PASSED [0.0617s]                                                                [  0%]
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_int PASSED [0.0553s]                                                                   [  1%]
test/inductor/test_torchinductor.py::SweepInputsCpuTest::test_cpu_broadcast1_strided PASSED [0.0586s]
...
```

### Versions

Observed on all platforms including CI jobs.

cc @ezyang @gchanan @zou3519 @kadeng @seemethere @malfet @pytorch/pytorch-dev-infra @msaroufim @wconstab @bdhirsh @anijain2305

### PR DESCRIPTION
**Pull request by malfet**
This is just follows the standard practice for CI, when one specifies `PYTORCH_TESTING_DEVICE_ONLY_FOR=cuda`, only tests targeting the device should be run

Do it by refactoring part of `instantiate_device_type_tests` into `get_desired_device_type_test_bases` and using it from test_torchinductor.py to skip CPU tests

Fixes https://github.com/pytorch/pytorch/issues/115423

cc @seemethere @pytorch/pytorch-dev-infra @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @voznesenskym @penguinwu @EikanWang @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: torch/testing/_internal/common_device_type.py [hunk 1]
```diff
@@ -722,12 +697,38 @@ def split_if_not_empty(x: str):
     env_only_for = split_if_not_empty(os.getenv(PYTORCH_TESTING_DEVICE_ONLY_FOR_KEY, ''))
     env_except_for = split_if_not_empty(os.getenv(PYTORCH_TESTING_DEVICE_EXCEPT_FOR_KEY, ''))
 
-    desired_device_type_test_bases = filter_desired_device_types(desired_device_type_test_bases,
-                                                                 env_except_for, env_only_for)
+    return filter_desired_device_types(desired_device_type_test_bases, env_except_for, env_only_for)
+
 
 
+# Adds 'instantiated' device-specific test cases to the given scope.
```
Comment (seemethere) [inline torch/testing/_internal/common_device_type.py:704]: Any particular reason for moving this function? Doesn't look like there's any functional changes here.

**Comment by kit1980** [comment]
Are submodules updates intentional?

**Comment by malfet** [comment]
> Are submodules updates intentional?

Thank you, no they were not, rebased

**Review by seemethere — APPROVED** [review]

**Pull request closed without merging**